client/components/empty-content/index.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { H } from 'layout/section';
+import ImageAsset from 'components/image-asset';
+
+/**
+ * A component to be used when there is no data to show.
+ * It can be used as an opportunity to provide explanation or guidance to help a user progress.
+ */
+class EmptyContent extends Component {
+	renderIllustration() {
+		const { illustrationWidth, illustrationHeight, illustration } = this.props;
+		return (
+			<ImageAsset
+				alt=""
+				src={ illustration }
+				width={ illustrationWidth }
+				height={ illustrationHeight }
+				className="woocommerce-empty-content__illustration"
+			/>
+		);
+	}
+
+	renderActionButtons( type ) {
+		const actionLabel =
+			'secondary' === type ? this.props.secondaryActionLabel : this.props.actionLabel;
+		const actionURL = 'secondary' === type ? this.props.secondaryActionURL : this.props.actionURL;
+		const actionCallback =
+			'secondary' === type ? this.props.secondaryActionCallback : this.props.actionCallback;
+
+		if ( actionURL && actionCallback ) {
+			return (
+				<Button
+					className="woocommerce-empty-content__action"
+					isPrimary
+					onClick={ actionCallback }
+					href={ actionURL }
+				>
+					{ actionLabel }
+				</Button>
+			);
+		} else if ( actionURL ) {
+			return (
+				<Button className="woocommerce-empty-content__action" isPrimary href={ actionURL }>
+					{ actionLabel }
+				</Button>
+			);
+		} else if ( actionCallback ) {
+			return (
+				<Button className="woocommerce-empty-content__action" isPrimary onClick={ actionCallback }>
+					{ actionLabel }
+				</Button>
+			);
+		}
+
+		return null;
+	}
+
+	renderActions() {
+		const { actionLabel, secondaryActionLabel } = this.props;
+		return (
+			<div className="woocommerce-empty-content__actions">
+				{ actionLabel && this.renderActionButtons( 'primary' ) }
+				{ secondaryActionLabel && this.renderActionButtons( 'secondary' ) }
+			</div>
+		);
+	}
+
+	render() {
+		const { title, message, illustration } = this.props;
+		return (
+			<div className={ classnames( 'woocommerce-empty-content', this.props.className ) }>
+				{ illustration && this.renderIllustration() }
+				{ title ? <H className="woocommerce-empty-content__title">{ title }</H> : null }
+				{ message ? <p className="woocommerce-empty-content__message">{ message }</p> : null }
+
+				{ this.renderActions() }
+			</div>
+		);
+	}
+}
+
+EmptyContent.propTypes = {
+	/**
+	 * The title to be displayed.
+	 */
+	title: PropTypes.string.isRequired,
+	/**
+	 * An additional message to be displayed.
+	 */
+	message: PropTypes.string,
+	/**
+	 * The url string of an image path. Prefix with `/` to load an image relative to the plugin directory.
+	 */
+	illustration: PropTypes.string.isRequired,
```

Comment:
What do you think about adding `empty-content.svg` as a default image here, and have this not be required? Do you think there's a case for having EmptyContent without an image?

Reply:
I'm fine with that, and it's what the Calypso version does actually. I think most of the time we'll probably want an image but I could see wanting just the text display in some circumstances.